src/test/scala/firrtl_interpreter/ConcreteSpec.scala

```diff
@@ -275,32 +275,109 @@ class ConcreteSpec extends FlatSpec with Matchers {
   behavior of "not"
 
   it should "flip bits of UInts" in {
+
+    def bitwiseXorReduction(s1: String, s2: String): Boolean = {
+      s1.zip(s2).forall {
+        case ('0', '1') => true
+        case ('1', '0') => true
+        case _          => false
+
+      }
+    }
+
     for(width <- IntWidthTestValuesGenerator(-MaxWidth, MaxWidth)) {
       if (width < -1 || width > 1) {
-        val bitString1 = (0 until width.abs-1).map(x => (x % 2).toString).mkString
-        val bitString2 = (0 until width.abs-1).map(x => ((x + 1) % 2).toString).mkString
-
-        for(sign <- Array(-1, 1)) {
-          val bigInt = sign*BigInt(bitString1, 2)
-          val si = ConcreteSInt(bigInt, width.abs)
-          val topUintBit1 = if (bigInt < 0) "1" else "0"
-          val topUintBit2 = if (bigInt < 0) "0" else "1"
-
-          val ui       = ConcreteUInt(BigInt(topUintBit1 + bitString1, 2), width.abs)
-          val expected = ConcreteUInt(BigInt(topUintBit2 + bitString2, 2), width.abs)
-
-//          println(s"width $width si     ${si.asBinaryString}")
-//          println(s"width $width ui     ${ui.asBinaryString}")
-//          println(s"width $width si.ex  ${expected.asBinaryString}")
-//          println(s"width $width si.not ${si.not.asBinaryString}")
-//          println()
-
-          si.not should be (expected)
-          ui.not should be (expected)
+        val bitString1 = (0 until width.abs).map(x => (x % 2).toString).mkString
+        val bitString2 = (0 until width.abs).map(x => ((x + 1) % 2).toString).mkString
+
+        for(bitString <- Seq(bitString1, bitString2)) {
+
+          val u1 = ConcreteUInt(BigInt(bitString, 2), width.abs)
+
+          bitwiseXorReduction(u1.toBinaryString, u1.not.toBinaryString) should be (true)
+
+          val s1 = u1.asSInt
+
+          bitwiseXorReduction(s1.toBinaryString, s1.not.toBinaryString) should be (true)
+
+          println(f"u1     $u1%60s ${u1.toBinaryString}")
+          println(f"u1.not ${u1.not}%60s ${u1.not.toBinaryString}")
+
+          println(f"s1     $s1%60s ${s1.toBinaryString}")
+          println(f"s1.not ${s1.not}%60s ${s1.not.toBinaryString}")
         }
       }
     }
   }
+
+  it should "invert an SInt, per: firrtl:issue-122" in {
```

Comment:
Last thing: just so that the linkage isn't lost, either `firrtl-interpreter:issue-122` or `firrtl:issue-753`.